packages/core/src/parser/Domain.ne

```diff
@@ -83,21 +83,23 @@ statements
     |  _ statement _c_ nl statements {% d => [d[1], ...d[4]] %}
 
 statement 
-  -> type        {% id %}
+  -> type_decl   {% id %}
   |  predicate   {% id %}
   |  function    {% id %}
   |  constructor_decl {% id %}
   |  prelude     {% id %}
   |  notation    {% id %}
   |  subtype     {% id %}
 
-type -> "type" __ identifier (_ "(" _ type_params _ ")"):? {%
-  ([typ, , name, ps]): TypeDecl => {
+# not to be confused with `type`, defined below
+type_decl -> "type" __ identifier (_ "(" _ type_params _ ")"):? (_ "<:" _ sepBy1[type, ","]):? {%
+  ([typ, , name, ps, sub]): TypeDecl => {
     const params = ps ? ps[3] : [];
+    const superTypes = sub ? sub[3] : [];
     return { 
-      ...nodeData([name, ...params]),
+      ...nodeData([name, ...params, ...superTypes]),
       ...rangeBetween(typ, name),
```

Comment:
@wodeni Is it OK that this range remains the same, or do I need to change it? I see that it already doesn't account for `params`, so I assumed it also shouldn't account for `superTypes`, but I'm not entirely sure that's correct.

Reply:
Makes sense; thanks for reviewing!